soc/intel/intel_adsp/tools/cavstool.py

```diff
@@ -249,14 +269,28 @@ def map_regs():
     global bar4_mmap
     (bar4_mem, bar4_mmap) = bar_map(pcidir, 4)
     dsp = Regs(bar4_mem)
-    dsp.ADSPCS         = 0x00004
-    dsp.HIPCTDR        = 0x00040 if cavs15 else 0x000c0
-    dsp.HIPCTDA        =                        0x000c4 # 1.8+ only
-    dsp.HIPCTDD        = 0x00044 if cavs15 else 0x000c8
-    dsp.HIPCIDR        = 0x00048 if cavs15 else 0x000d0
-    dsp.HIPCIDA        =                        0x000d4 # 1.8+ only
-    dsp.HIPCIDD        = 0x0004c if cavs15 else 0x000d8
-    dsp.SRAM_FW_STATUS = 0x80000 # Start of first SRAM window
+    if ace15 or ace20:
+        dsp.HFDSSCS = 0x1000
+        dsp.HFPWRCTL = 0x1d18
+        dsp.HFPWRSTS = 0x1d1c
+        dsp.DSP2CXCTL_PRIMARY = 0x178d04
+        dsp.HFIPCXTDR = 0x73200
+        dsp.HFIPCXTDA = 0x73204
+        dsp.HFIPCXIDR = 0x73210
+        dsp.HFIPCXIDA = 0x73214
+        dsp.HFIPCXCTL = 0x73228
+        dsp.HFIPCXTDDY = 0x73300
+        dsp.HFIPCXIDDY = 0x73380
+        dsp.SRAM_FW_STATUS = WINDOW_BASE_ACE
+    else:
+        dsp.ADSPCS         = 0x00004
+        dsp.HIPCTDR        = 0x00040 if cavs15 else 0x000c0
+        dsp.HIPCTDA        =                        0x000c4 # 1.8+ only
+        dsp.HIPCTDD        = 0x00044 if cavs15 else 0x000c8
+        dsp.HIPCIDR        = 0x00048 if cavs15 else 0x000d0
+        dsp.HIPCIDA        =                        0x000d4 # 1.8+ only
+        dsp.HIPCIDD        = 0x0004c if cavs15 else 0x000d8
+        dsp.SRAM_FW_STATUS = WINDOW_BASE # Start of first SRAM window
```

Comment:
Nitpick: SRAM_FW_STATUS can be mapped once outside the if based on the result of adsp_mem_window_config().  Saves a line, maybe clearer since the feature is identical.

Reply:
I was about to change, but let me keep this as is. This is actually not quite correct for the ACE family, but the problem is already in the old code (and it sort of works), but let me return to this in a follow-up PR (to align with changed done in Linux SOF driver for ACE family).